### DISCUSSION TITLE
Deleting all messages from a specific user

### DISCUSSION THREAD

**Discussion by Pranay0147** [python-telegram-bot/python-telegram-bot#3586] [Q&A] [answered]
when replied to a message in a group with "deleteall" command the bot must be able to delete all the previous messages of the user atleast for the past 48 hours how can i do this?

**Comment by Poolitzer** [comment, marked as answer]
You need to store all incoming message ids from the user and iterate over that list

**Reply by Pranay0147** [reply]
is there any way to do that without storing,  kind of like getting history/previous chat from the group and iterating over it?
Because storing limits the bot to execute the command if it is newly added to a group.

Thank you.

**Reply by Poolitzer** [reply]
No

**Reply by Sanix-Darker** [reply]
Hey there, i really like the project and i was just wondering, it could be great to have a way to clean everything 😅 ... the idea of tracking each message_id is good but just like the `bot.delete...` the API should provide a `bot.deleteAll`... that means we could track inside the lib all message for a given chat_id... ?

As @sulehacker1 suggested, i can make a small PR to handle this feature request
What do you think @Poolitzer ?

Best.

**Reply by Poolitzer** [reply]
Hi, that is not going to work. A bot can only get updates from messages once it has been added to a chat (and it doesn't get all service messages iirc) and only in a 48 hour time window, so delete_all probably would never delete all messages in a chat.

If you still want to implement this for the people who still want to use this despite this, feel free to make a module at [ptbcontrib](https://github.com/python-telegram-bot/ptbcontrib)

**Comment by sulehacker1** [comment]
> وقتی به یک پیام در یک گروه با دستور "حذف همه" پاسخ داده می شود، ربات باید بتواند تمام پیام های قبلی کاربر را حداقل در 48 ساعت گذشته حذف کند چگونه می توانم این کار را انجام دهم؟



> when replied to a message in a group with "deleteall" command the bot must be able to delete all the previous messages of the user atleast for the past 48 hours how can i do this?

There is a way
Messages in a chat are numbered in a staggered manner
The first message: 1
The second message: 2

You can get the ID of the message when the user sent the command to delete the message and go back in reverse
Pay attention that you must use "try, except".
Because some of the messages may have been deleted manually and the robot may not be able to delete them

**Reply by Sanix-Darker** [reply]
thanks for this, does that means, when i reach 0, i cleaned everything ?
